python/paddle/nn/functional/flash_attention.py

```diff
@@ -1251,3 +1251,103 @@ def flash_attention_with_sparse_mask(
         return outputs[0]
     else:
         return outputs
+
+
+def calc_reduced_attention_scores(
+    query: paddle.Tensor, key: paddle.Tensor, softmax_lse: paddle.Tensor
+) -> paddle.Tensor:
+    r"""
+    The equation is:
+
+    .. math::
+
+        result=reduce_sum(softmax(\frac{ Q * K^T }{\sqrt{d}}), dim=-2)
+
+    Warning:
+        This API only supports inputs with dtype float16 and bfloat16.
+
+    Args:
+        query(Tensor): The query tensor in the Attention module.
+                        4-D tensor with shape:
+                        [batch_size, seqlen_q, num_heads, head_dim].
+                        The dtype can be float16 or bfloat16.
+        key(Tensor): The key tensor in the Attention module.
+                        4-D tensor with shape:
+                        [batch_size, seqlen_k, num_heads, head_dim].
+                        The dtype can be float16 or bfloat16.
+        softmax_lse(Tensor): The logsumexp of each row returned by _C_ops.flash_attn().
+                        3-D tensor with shape:
+                        [batch_size, num_heads, seqlen_q].
+                        The dtype is float32.
+    Returns:
+        reduced_attention_scores(Tensor), The reduce sum of attention scores across seqlen_q.
+                    4-D tensor with shape: [batch_size, num_heads, 1, seqlen_k].
+                    The dtype is float32.
```

Comment:
```suggestion
    Returns:
        reduced_attention_scores(Tensor), The reduce sum of attention scores across seqlen_q.
        4-D tensor with shape ``[batch_size, num_heads, 1, seqlen_k]``. The dtype is float32.
```
官网渲染出了点问题
![image](https://github.com/PaddlePaddle/Paddle/assets/70642955/750006c2-3080-41ae-839f-d23183ab4d53)